### PR TITLE
Make psutil dependency optional

### PR DESCRIPTION
**Pull request by youngmit**
MacOS users have trouble getting psutil to work, and it is only needed
to do memory profiling, which is not a necessity. This makes the psutil
package optional, and hobbles most of the MemoryProfiler functionality
if it isn't found. Removing the MemoryProfiler entirely in the absence
of psutil might be a better long-term goal, but it is rather intertwined
with reports and the standard operator and difficult to remove.

### CODE REVIEW

**Comment by youngmit** [comment]
Do we think it's worth it to add test environments for testing without extra features?

File: armi/bookkeeping/memoryProfiler.py [hunk 1]
```diff
@@ -42,10 +42,17 @@
...
 import tabulate
-import gc
+from typing import Optional
+
+try:
+    import psutil
+
+    _havePsutil = True
+except ImportError:
+    _havePsutil = False
```
Comment (ntouran) [inline armi/bookkeeping/memoryProfiler.py:55]: add docs explaining why this is this way to future code readers

File: armi/bookkeeping/memoryProfiler.py [hunk 2]
```diff
@@ -502,26 +509,31 @@ def invokeHook(self):
 class SystemAndProcessMemoryUsage(object):
     def __init__(self):
         self.nodeName = armi.MPI_NODENAME
-        self.percentNodeRamUsed = psutil.virtual_memory().percent
-        self.processMemoryInMB = psutil.Process().memory_info().rss / (1012.0 ** 2)
+        self.percentNodeRamUsed: Optional[float] = None
+        self.processMemoryInMB: Optional[float] = None
+        if _havePsutil:
```
Comment (ntouran) [inline armi/bookkeeping/memoryProfiler.py:514]: else: 
warning: you don't have psutil so can't do anything here

**Review by ntouran — APPROVED** [review]
pls add comments explaining the situation.

**Pull request merged**